packages/admin/admin-rte/src/core/xml/transformStateToXml.ts

```diff
@@ -0,0 +1,162 @@
+    [INLINE_STYLE.BOLD]: { element: "inline" },
+    [INLINE_STYLE.ITALIC]: { element: "inline" },
+    [INLINE_STYLE.STRIKETHROUGH]: { element: "inline" },
+    [INLINE_STYLE.UNDERLINE]: { element: "inline" },
+};
+
+/* 
+    escapes special characters in the text content to their HTML/XML entities
+*/
+function encodeContent(text: string): string {
+    return text.split("&").join("&amp;").split("<").join("&lt;").split(">").join("&gt;").split("\xA0").join("&nbsp;").split("\n").join(`<br>\n`);
+}
+
+class MarkupGenerator {
+    blocks: ContentBlock[] = [];
+    contentState: ContentState | undefined;
+    output: string[] = [];
+    currentBlock = 0;
+    indentLevel = 0;
+    totalBlocks = 0;
+    inlineStyles = DEFAULT_STYLE_MAP;
+    styleOrder: string[] = DEFAULT_STYLE_ORDER;
+    indent = "  ";
+    counter = 1;
+
+    constructor(contentState?: ContentState) {
+        this.contentState = contentState;
+    }
+
+    generate(): string[] {
+        if (!this.contentState) {
+            return [];
+        }
+
+        this.output = [];
+        this.blocks = this.contentState.getBlocksAsArray();
+        this.totalBlocks = this.blocks.length;
+        this.currentBlock = 0;
+
+        while (this.currentBlock < this.totalBlocks) {
+            this.processBlock(this.contentState);
+        }
+
+        return this.output.filter((content) => content !== "" && content !== "\n");
+    }
+
+    processBlock(contentState: ContentState) {
+        const block = this.blocks[this.currentBlock];
+
+        const content = this.renderBlockContent(block, contentState);
+
+        this.output.push(content);
+
+        this.currentBlock += 1;
+        this.output.push(`\n`);
+    }
+
+    renderBlockContent(block: ContentBlock, contentState: ContentState): string {
+        let text = block.getText();
+
+        let currentLinkId = 0;
+
+        if (text === "") {
+            return "";
+        }
+
+        text = this.preserveWhitespace(text);
+
+        // getting a list including all styles and entites for every single character
+        const charMetaList: CharacterMetaList = block.getCharacterList();
+
+        // divides the information about style and entities of each character into ranges
+        const entityPieces = getEntityRanges(text, charMetaList);
+
+        return entityPieces
+            .map(([entityKey, stylePieces]) => {
+                const content = stylePieces
+                    .map(([text, styleSet]) => {
+                        let content = encodeContent(text);
+                        for (const styleName of this.styleOrder) {
+                            const currentStyle = styleSet.find((style) => style.style === styleName);
+
+                            if (currentStyle) {
+                                let { element } = this.inlineStyles[styleName];
+                                if (element == null) {
+                                    element = "span";
+                                }
```

Comment:
For custom inline styles, the resulting tag would be inline 🤔

Reply:
further adaptions are needed to use custom inline styles, but I think this should be considered in a next PR